amazon/ion/ioncmodule.c

```diff
@@ -0,0 +1,1369 @@
+            break;
+        }
+        case tid_BOOL_INT:
+        {
+            BOOL bool_value;
+            IONCHECK(ion_reader_read_bool(hreader, &bool_value));
+            py_value = PyBool_FromLong(bool_value);
+            ion_nature_constructor = _ionpybool_fromvalue;
+            break;
+        }
+        case tid_INT_INT:
+        {
+            // TODO add ion-c API to return an int64 if possible, or an ION_INT if necessary
+            ION_INT ion_int_value;
+            IONCHECK(ion_int_init(&ion_int_value, hreader));
+            IONCHECK(ion_reader_read_ion_int(hreader, &ion_int_value));
+            SIZE int_char_len, int_char_written;
+            IONCHECK(ion_int_char_length(&ion_int_value, &int_char_len));
+            char* ion_int_str = (char*)PyMem_Malloc(int_char_len + 1); // Leave room for \0
+            err = ion_int_to_char(&ion_int_value, (BYTE*)ion_int_str, int_char_len, &int_char_written);
+            if (err) {
+                PyMem_Free(ion_int_str);
+                IONCHECK(err);
+            }
+            if (int_char_len < int_char_written) {
+                PyMem_Free(ion_int_str);
+                _FAILWITHMSG(IERR_BUFFER_TOO_SMALL, "Not enough space given to represent int as string.");
+            }
+            py_value = PyLong_FromString(ion_int_str, NULL, 10);
+            PyMem_Free(ion_int_str);
+
+            ion_nature_constructor = _ionpyint_fromvalue;
+            break;
+        }
+        case tid_FLOAT_INT:
+        {
+            double double_value;
+            IONCHECK(ion_reader_read_double(hreader, &double_value));
+            py_value = Py_BuildValue("d", double_value);
+            ion_nature_constructor = _ionpyfloat_fromvalue;
+            break;
+        }
+        case tid_DECIMAL_INT:
+        {
+            ION_DECIMAL decimal_value;
+            IONCHECK(ion_reader_read_ion_decimal(hreader, &decimal_value));
+            char dec_str[DEC_NUMBER_MAX_LEN]; // DECQUAD_String is only 43, which fails some tests.
+            IONCHECK(ion_decimal_to_string(&decimal_value, dec_str));
+            c_decstr_to_py_decstr(&dec_str);
+            py_value = PyObject_CallFunction(_decimal_constructor, "s", dec_str);
+            ion_nature_constructor = _ionpydecimal_fromvalue;
+            break;
+        }
+        case tid_TIMESTAMP_INT:
+        {
+            IONCHECK(ionc_read_timestamp(hreader, &py_value));
+            ion_nature_constructor = _ionpytimestamp_fromvalue;
+            break;
+        }
+        case tid_SYMBOL_INT:
+        {
+            emit_bare_values = FALSE; // Symbol values must always be emitted as IonNature because of ambiguity with string.
+            ION_STRING string_value;
+            IONCHECK(ion_reader_read_string(hreader, &string_value));
+            ion_nature_constructor = _ionpysymbol_fromvalue;
+            py_value = ion_string_to_py_symboltoken(&string_value);
+            break;
```

Comment:
Does this work for symbols with unknown text?

Reply:
As discussed offline, we will use pure python implementation to handle `catalog` and `import` options and support more options incrementally.